openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/utils/TemplateUtils.java

```diff
@@ -72,7 +75,12 @@ public static String readTemplateUrl(Template template, CubeOpenShiftConfigurati
 
         if (templateUrl != null) {
             String url = resolver.resolve(templateUrl);
-            templateUrl = ResourceResolver.resolve(url).toString();
+
+            try {
+                templateUrl = new URL(url).toString();
+            } catch (MalformedURLException ex) {
+                throw new IllegalArgumentException(ex);
+            }
```

Comment:
The only reason we were "resolving" it earlier was to actually wrap it in URL and return string representation of it. Now we return `InputStream` in the `resolve` method reading it earlier. I decided to just validate the URL in the old form, as you can see.